crabs/detection_tracking/inference_model.py

```diff
@@ -136,50 +153,173 @@ def prep_csv_writer(self):
+
+        Parameters:
+        -----------
+        gt_boxes_list : List[list]
+            List of ground truth bounding boxes for each frame.
+        tracked_boxes_list : List[list]
+            List of tracked bounding boxes for each frame.
+        iou_threshold : float
+            The IoU threshold used to determine matches between ground truth and tracked boxes.
+
+        Returns:
+        --------
+        float:
+            The computed MOTA (Multi-Object Tracking Accuracy) score for the tracking performance.
         """
+        from crabs.detection_tracking.detection_utils import evaluate_mota
 
-        # Bounding box geometry
-        xmin, ymin, xmax, ymax, id = bbox
-        width_box = int(xmax - xmin)
-        height_box = int(ymax - ymin)
-
-        # Add to csv
-        csv_writer.writerow(
-            (
-                frame_name,
-                frame.size,
-                '{{"clip":{}}}'.format("123"),
-                1,
-                0,
-                '{{"name":"rect","x":{},"y":{},"width":{},"height":{}}}'.format(
-                    xmin, ymin, width_box, height_box
-                ),
-                '{{"track":{}}}'.format(id),
+        mota_values = []
+        prev_frame: Union[Optional[List[List[float]]], None] = None
+        for gt_boxes, tracked_boxes in zip(gt_boxes_list, tracked_boxes_list):
+            mota = evaluate_mota(
+                gt_boxes, tracked_boxes, iou_threshold, prev_frame
             )
-        )
+            mota_values.append(mota)
+            # Update previous frame IDs for the next iteration
+            prev_frame = [list(box) for box in tracked_boxes]
+
+        return mota_values
+
+    def get_prediction(self, frame: np.ndarray) -> torch.Tensor:
+        """
+        Get prediction from the trained model for a given frame.
+
+        Parameters:
+        -----------
+        frame : np.ndarray
+            The input frame for which prediction is to be obtained.
+
+        Returns:
+        --------
+        torch.Tensor:
+            The prediction tensor from the trained model.
+        """
+        transform = transforms.Compose([transforms.ToTensor()])
+        img = transform(frame).to(self.args.accelerator)
+        img = img.unsqueeze(0)
+        return self.trained_model(img)
+
+    def update_tracking(self, prediction: dict) -> List[List[float]]:
+        """
+        Update the tracking system with the latest prediction.
+
+        Parameters:
+        -----------
+        prediction : dict
+            Dictionary containing predicted bounding boxes, scores, and labels.
+
+        Returns:
+        --------
+        List[List[float]]:
+            List of tracked bounding boxes after updating the tracking system.
+        """
+        pred_sort = self.prep_sort(prediction)
+        tracked_boxes = self.sort_tracker.update(pred_sort)
+        self.tracked_list.append(tracked_boxes)
+        return tracked_boxes
+
+    def handle_output(
```

Comment:
```suggestion
    def save_required_output(
```
maybe a bit more explicit?